src/app/panels/terminal.js

```diff
@@ -103,14 +105,27 @@ class Terminal {
 
     self._jsSandboxContext = {}
     self._jsSandboxRegistered = {}
+
+    self.externalApi = this.api()
+    opts.appManager.init([self.externalApi])
+    opts.appManager.activateRequestAndNotification(self.externalApi)
+
     if (opts.shell) self._shell = opts.shell
     register(self)
   }
-
   focus () {
     if (this._view.input) this._view.input.focus()
   }
-
+  get profile () {
+    return {
+      displayName: 'terminal',
+      name: 'terminal',
+      methods: [],
+      events: [],
+      description: ' - ',
+      required: false
+    }
+  }
```

Comment:
Create const variable outside the class, and add it inside the super() : 
```javascript
const profile = {
      displayName: 'terminal',
      name: 'terminal',
      methods: [],
      events: [],
      description: ' - ',
      required: false
}
....
constructor(opts, api) {
  super(profile)
  ...
}
```

Reply:
please always use uppercase for displayName. terminal->Terminal